spec/lib/remote_console/web_mks_legacy_spec.rb

```diff
@@ -0,0 +1,25 @@
+describe RemoteConsole::WebMKSLegacy do
+  let(:record) { FactoryGirl.create(:system_console, :url => '/12345') }
+  subject { described_class.new(record, nil) }
+
+  before do
+    ssl = double
+    allow(OpenSSL::SSL::SSLSocket).to receive(:new).and_return(ssl)
+    allow(OpenSSL::X509::Certificate).to receive(:new)
+    allow(OpenSSL::PKey::RSA).to receive(:new)
+    allow(File).to receive(:open)
+    allow(ssl).to receive(:sync_close=)
+    allow(ssl).to receive(:connect)
```

Comment:
Couldn't we just mock setup_ssl function on SSLSocket and avoid all the cert issues? I mean, that's why we introduced it in first place ;)

```ruby
allow(subject).to receive(:setup_ssl).and_return(double('ssl', :connect => nil))
```

Reply:
No, because the subject is being created before the mocking can be done and I'm strongly against using `allow_any_instance_of`.